### PR TITLE
Specify CONNECT error handling bidirectionally

### LINKED ISSUE #4939: CONNECT might be under-specified
From [a discussion on the use of extended CONNECT for thewebsocketprotocol](https://github.com/httpwg/http-extensions/issues/1701)...

[This text](https://quicwg.org/base-drafts/draft-ietf-quic-http.html#section-4.2-10):

> Correspondingly, if a proxy detects an error with the stream or the QUIC connection, it MUST close the TCP connection. If the underlying TCP implementation permits it, the proxy SHOULD send a TCP segment with the RST bit set.

...could be improved in h3. Not the TCP RST part, but the bit that leads up to it.  Otherwise, we might not get the right behaviour.  If the proxy just closes the connection in response to RESET_STREAM, it might then just end the stream (with a FIN bit), which would be wrong.

Perhaps instead:

> Correspondingly, if a proxy detects an error with the stream or the QUIC connection, it MUST close the TCP connection. Similarly, if the proxy receives a QUIC RESET_STREAM frame or STOP_SENDING frame, it MUST close the TCP connection. A proxy that receives RESET_STREAM or STOP_SENDING SHOULD send the same frame in response in order to ensure that both directions of the stream are cancelled. In all these cases, if the underlying TCP implementation [...]

Somewhat substantive, unfortunately, and very, very, very late.  I'm happy to wait and file an erratum.

### PR DESCRIPTION
**Pull request by MikeBishop**
Fixes #4939

### CODE REVIEW

**Review by martinthomson — APPROVED** [review]
Better than my text :)

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -793,10 +793,15 @@ data from the target of the CONNECT.
...
 with the RST bit set, as a stream error of type H3_CONNECT_ERROR; see
-{{errors}}.  Correspondingly, if a proxy detects an error with the stream or the
-QUIC connection, it MUST close the TCP connection.  If the underlying TCP
-implementation permits it, the proxy SHOULD send a TCP segment with the RST bit
-set.
+{{errors}}.
+
+Correspondingly, if a proxy detects an error with the stream or the QUIC
+connection, it MUST close the TCP connection.  Similarly, if the proxy receives
+a QUIC RESET_STREAM frame or STOP_SENDING frame for the stream, it MUST close
```
Comment (DavidSchinazi) [inline draft-ietf-quic-http.md:800]: It makes sense for RESET_STREAM to RST the TCP connection, but STOP_SENDING could be safely ignored because there's no equivalent in TCP?

**Review by kazuho — COMMENTED** [review]
The text looks technically correct to me, though it was a bit of a surprise for me to see text discussing QUIC v1 frames.

IIRC, up until now, we have talked about concepts (e.g., stream resets) rather than the QUIC v1 frames that covey such signals. Is the diversion intentional?

**Comment by MikeBishop** [comment]
@kazuho, that's a good point.  Let me reword it to use the operation terms rather than the v1 frame names.

**Review by kazuho — APPROVED** [review]
